test/sharness/t0051-object.sh

```diff
@@ -145,6 +145,16 @@ test_object_cmd() {
 		echo QmUNLLsPACCz1vLxQVkXqqLX5R1X345qqfHbsf67hvA3Nn > rmlink_exp &&
 		test_cmp rmlink_exp rmlink_output
 	'
+
+	test_expect_success "object patch --create works" '
+		OUT=$(ipfs object patch --create $EMPTY add-link a/b/c $FILE)
+	'
+
+	test_expect_success "result looks good" '
+		ipfs cat $OUT/a/b/c > p2_hwfile &&
+		test_cmp hwfile p2_hwfile
+	'
```

Comment:
test edge + error cases too:

```
a
a/b
a/b/b/b/b
/ 
```
and one where the link already exists on the object.

recommend doing this sort of thing with a function:

```sh
test_patch_create_path() {
  root=$1
  name=$2
  target=$3

  test_expect_success "object patch --create works" ' ... '

  test_expect_success "result looks good" ' ... '

}

test_patch_create_path $EMPTY a/b/c $FILE

test_patch_create_path $EMPTY a $FILE

... 
```